modules/ding_nodelist/ding_nodelist.module

```diff
@@ -0,0 +1,2477 @@
+    'file' => 'ding_nodelist.admin.inc',
+  );
+
+  $items['admin/config/ding/ding_nodelist/settings'] = array(
+    'title' => 'Settings',
+    'description' => 'Configure general settings.',
+    'tab_parent' => 'admin/config/ding/ding_nodelist',
+    'type' => MENU_DEFAULT_LOCAL_TASK,
+    'file' => 'ding_nodelist.admin.inc',
+  );
+
+  $items['admin/config/ding/ding_nodelist/templates'] = array(
+    'title' => 'Templates',
+    'description' => 'Manage content templates.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ding_nodelist_templates_form'),
+    'access arguments' => array('configure nodelist'),
+    'type' => MENU_LOCAL_TASK,
+    'file' => 'ding_nodelist.admin.inc',
+  );
+
+  $items['admin/config/ding/ding_nodelist/trim_content'] = array(
+    'title' => 'Trim content',
+    'description' => 'Define title and teaser trim parameters for each widget',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ding_nodelist_trim_content'),
+    'access arguments' => array('configure nodelist'),
+    'type' => MENU_LOCAL_TASK,
+    'file' => 'ding_nodelist.admin.inc',
+  );
+
+  $items['ding_nodelist/autocomplete/%'] = array(
+    'page callback' => 'ding_nodelist_autocomplete',
+    'page arguments' => array(2),
+    'access arguments' => array('access content'),
+    'type' => MENU_CALLBACK,
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_permission().
+ */
+function ding_nodelist_permission() {
+  return array(
+    'configure nodelist' => array(
+      'title' => t('Configure nodelist'),
+      'description' => t('Configure the behavior of nodelist'),
+    ),
+    'nodelist cache settings' => array(
+      'title' => t('Nodelist cache'),
+      'description' => t('Manage nodelist cache settings'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_ctools_plugin_directory().
+ */
+function ding_nodelist_ctools_plugin_directory($module, $plugin) {
+  // we'll be nice and limit scandir() calls.
+  if ($module == 'ctools' && $plugin == 'content_types') {
+    return 'plugins/content_types';
+  }
+}
+
+/**
+ * Implements hook_ajax_render_alter().
+ *
+ * This method is a workaround to impossiblity
+ * of telling Drupal the order of CSS files in order to
+ * cascade styles properly. See the link.
+ *
+ * @link http://drupal.org/node/561858#comment-3171846
+ *
+ * @global boolean $_ding_nodelist_ipe_mode
+ */
+function ding_nodelist_ajax_render_alter(&$commands = array()) {
+  global $_ding_nodelist_ipe_mode, $_ding_nodelist_ipe_redirect;
+
+  if ($_ding_nodelist_ipe_mode) {
+    $path = drupal_get_path('module', 'ding_nodelist') . "/css/ipe.css";
+    $styles = '<style type="text/css" media="all">@import url("' . $path . '");</style>';
+    $commands[] = ajax_command_append('head', $styles);
+  }
+
+  if ($_ding_nodelist_ipe_redirect) {
+    ctools_include('ajax');
+    ctools_add_js('ajax-responder');
+    $commands[] = ctools_ajax_command_redirect($_ding_nodelist_ipe_redirect);
```

Comment:
This final part of `ding_nodelist_ajax_render_alter()` does not seem to relate to the problem mentioned in the method docblock.

Please document it. Doing a redirect in a alter hook seems odd to me.

Also consider moving the part of the docblock dealing with the CSS problem into the function as an inline comment and above the code dealing with that specific problem.